app/javascript/components/footprints.vue

```diff
@@ -0,0 +1,111 @@
+<template lang="pug">
+.user-icons(v-if='!loaded')
+  .fa-solid.fa-spinner.fa-pulse
+  |
+  | ロード中
+.user-icons(v-else)
+  h3.user-icons__title
+    | 見たよ
+  ul.user-icons__items(
+    v-if='tenOrLessFootprints || (moreThanTenFoorptints && isDisplay)'
+  )
+    footprint(
+      v-for='footprint in limitedDisplayOnPage',
+      :key='footprint.key',
+      :footprint='footprint'
+    )
+  .user-icons__more(
+    v-if='moreThanTenFoorptints && isDisplay',
+    @click='showRemainingFootprints'
+  )
+    | その他{{ numberOfRemainingFootprints }}人
+  ul.user-icons__items(v-else-if='moreThanTenFoorptints && !isDisplay')
```

Comment:
v-ifだけでも動作は同じになりそうだと思いましたが、どうでしょうか。
v-else-ifはどこに掛かってるのか分かりづらいので、代替できるようならv-ifの方が良さそうです。
```suggestion
  ul.user-icons__items(v-if='moreThanTenFoorptints && !isDisplay')
```